sys/src/9/ufs/ffs/ffs_vfsops.c

```diff
@@ -905,28 +905,28 @@ ffs_mountfs (vnode *devvp, MountPoint *mp, thread *td)
 	memset(fs->fs_contigdirs, 0, size);
 	fs->fs_active = nil;
 	mp->mnt_data = ump;
-#if 0
-	mp->mnt_stat.f_fsid.val[0] = fs->fs_id[0];
+	// TODO HARVEY - maybe we won't need this
+	/*mp->mnt_stat.f_fsid.val[0] = fs->fs_id[0];
 	mp->mnt_stat.f_fsid.val[1] = fs->fs_id[1];
 	nmp = nil;
 	if (fs->fs_id[0] == 0 || fs->fs_id[1] == 0 ||
 	    (nmp = vfs_getvfs(&mp->mnt_stat.f_fsid))) {
 		if (nmp)
 			vfs_rel(nmp);
 		vfs_getnewfsid(mp);
-	}
-	mp->mnt_maxsymlinklen = fs->fs_maxsymlinklen;
-	MNT_ILOCK(mp);
+	}*/
+	//mp->mnt_maxsymlinklen = fs->fs_maxsymlinklen;
+	qlock(&mp->mnt_lock);
```

Comment:
MNT_ILOCK is a wrapper to lock the mutex on the mountpoint.  I take FreeBSDs mutex to be equivalent to our qlock, since neither are spinlocks.  Not sure how they'd behave with interrupts tbh.